Geometry/CaloGeometry/src/CaloGeometry.cc

```diff
@@ -16,7 +16,7 @@ unsigned int CaloGeometry::makeIndex(DetId::Detector det, int subdet, bool& ok)
     edm::LogWarning("CaloGeometry") << "Det:Subdet " << idet << ":" << subdet << " min|max Det " << kMinDet << ":"
                                     << kMaxDet << " min|max subdet 0:" << kMaxSub;
 
-  return ((det - kMinDet) * kNSubDets + subdet);
+  return ((det - static_cast<int>(kMinDet)) * kNSubDets + subdet);
```

Comment:
I think it would be better to define all these
https://github.com/cms-sw/cmssw/blob/a9682e8ca89d4942c6676bf8c5f0d2c6a70abb0e/Geometry/CaloGeometry/interface/CaloGeometry.h#L59-L66
as `static constexpr` constants.